### PR TITLE
[WFCORE-5873] Add tests for WFCORE-5861 InaccessibleObjectException w…

### PR DESCRIPTION
**Pull request by yersan**
…ith OpenJDK17 and embed-server

Jira issue: https://issues.redhat.com/browse/WFCORE-5873

### CODE REVIEW

**Comment by yersan** [comment]
JDK-17 job: https://ci.wildfly.org/viewLog.html?buildId=300458&buildTypeId=WildFlyCore_MainLinuxJdk17

**Comment by yersan** [comment]
The new JbossCliClientJarJPMSSettingsTestCase passed with JDK17 Job. This is ready for review.

**Comment by xstefank** [comment]
@bstansberry, this seems to be stuck for a while. Can you please look?

File: testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/JbossCliClientJarJPMSSettingsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    @Test
+    public void testManifestEntries() throws Exception {
+        JarFile clientJar = new JarFile(SERVER_HOME.resolve("bin").resolve("client").resolve("jboss-cli-client.jar").toString());
+        Manifest manifest = clientJar.getManifest();
+        Attributes mainAttribs = manifest.getMainAttributes();
+        String addOpens = mainAttribs.getValue("Add-Opens");
+        Assert.assertTrue("Unexpected value for Manifest Add-Opens. Current: " + addOpens,
+                addOpens.contains("java.base/java.util"));
+        Assert.assertTrue("Unexpected value for Manifest Add-Opens. Current: " + addOpens,
+                addOpens.contains("java.base/java.lang.invoke"));
```
Comment (bstansberry) [inline testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/JbossCliClientJarJPMSSettingsTestCase.java:75]: These can be a lot more extensive, i.e. what the jar contains now is:

Add-Exports: java.desktop/sun.awt java.naming/com.sun.jndi.ldap java.nam
 ing/com.sun.jndi.url.ldap java.naming/com.sun.jndi.url.ldaps
Add-Opens: java.base/java.lang java.base/java.lang.invoke java.base/java
 .lang.reflect java.base/java.io java.base/java.security java.base/java.
 util java.base/java.util.concurrent java.management/javax.management ja
 va.naming/javax.naming

If you pass maven prop embedding.jar.jpms.exports into surefire via -D you could parse it out and dynamically check for all of them.

**Review by bstansberry — APPROVED** [review]
I approved this as it good progress, but see my comment for potential further improvement.

File: testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/JbossCliClientJarJPMSSettingsTestCase.java (hunk 1 above)
Comment (yersan) [reply testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/JbossCliClientJarJPMSSettingsTestCase.java:75]: Thanks for reviewing it @bstansberry . Got it, I'll add a follow-up Jira for your suggestion, originally it was not my intention to verify all the settings are in the Jar.

The intention here was just to check if the required settings used by the test were satisfied to ensure we are not getting a false negative. The test case is only verifying a subset of required settings by executing something that requires them.

**Comment by bstansberry** [comment]
@yersan That's a good point, and leaving the two hard coded MANIFEST checks you have makes sense for that reason.

I don't think doing more is urgent. It would kind of just be a regression test in case we mess up how the manifest is built. Not a real strong one though, as the embedding.jar.jpms.exports prop is an implementation detail, so we'd be baking an assumption of how the impl works into a test.

**Pull request merged**